### ISSUE TITLE
Chat - Some emoji don't transfer from Slack to Chat Expensify

### ISSUE THREAD

**Issue by kbecciv** [Expensify/App#17353]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. Navigate to a chat app that supports emojis (For example: Slack).
2. Send a message containing multiple emojis and text.
3. Copy the message containing multiple emojis and text.
4. Switch to New Expensify.
5. Go to any conversation.
6. Paste the copied message into composer

## Expected Result:
Verify that emojis and text are correctly pasted in composer.

## Actual Result:
Some emoji don't transfer from Slack to Chat Expensify

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [x] Android / native
- [x] Android / Chrome
- [x] iOS / native
- [x] iOS / Safari
- [x] MacOS / Chrome / Safari
- [x] MacOS / Desktop

**Version Number:** 1.2.99.3

**Reproducible in staging?:** Yes

**Reproducible in production?:** Yes

**If this was caught during regression testing, add the test name, ID and link from TestRail:**

**Email or phone of affected tester (no customers):**

**Logs:** https://stackoverflow.com/c/expensify/questions/4856

**Notes/Photos/Videos:** Any additional supporting documentation

https://user-images.githubusercontent.com/93399543/231552183-3bb7b511-e2f2-4d96-84f9-33cb7e702818.mp4

**Expensify/Expensify Issue URL:**

**Issue reported by:** Applause - Internal Team

**Slack conversation:**

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by MelvinBot** [comment]
Triggered auto assignment to @strepanier03 (`Bug`), see https://stackoverflow.com/c/expensify/questions/14418 for more details.

**Comment by MelvinBot** [comment]
### Bug0 Triage Checklist ([Main S/O](https://stackoverflowteams.com/c/expensify/questions/14418))

- [ ] This "bug" occurs on a supported platform (ensure `Platforms` in OP are ✅)
- [ ] This bug is not a duplicate report (check [E/App issues](https://github.com/Expensify/App/issues) and [#expensify-bugs](https://expensify.slack.com/archives/C049HHMV9SM))
    - If it is, comment with a link to the original report, close the issue and add any novel details to the original issue instead
- [ ] This bug is reproducible using the reproduction steps in the OP. [S/O](https://stackoverflowteams.com/c/expensify/questions/14418#:~:text=The%20bug%20is%20reproducible)
    - If the reproduction steps are clear and you're unable to reproduce the bug, check with the reporter and QA first, then close the issue.
    - If the reproduction steps aren't clear and you determine the correct steps, please update the OP.
- [ ] This issue is filled out as thoroughly and clearly as possible
    - Pay special attention to the title, results, platforms where the bug occurs, and if the bug happens on staging/production.
- [ ] I have reviewed and subscribed to the linked Slack conversation to ensure Slack/Github stay in sync

**Comment by marcaaron** [comment]
I think it's not a bug as there is no way to possibly know what emojis exist in Slack since many are custom 🤔

**Comment by strepanier03** [comment]
Thank you @marcaaron I was feeling the same. Both Discord and Slack support custom emoji, I'm sure other chat apps do as well. 

If the emoji in Slack or Discord is one we have in Echat, then I think copy-pasting them from Slack/Discord to Echat should work. 

I'm going to do some testing and see if we generally do allow copy/pasting them from Slack to Echat if they exist in both systems.

**Comment by strepanier03** [comment]
So testing in Slack and Echat, I found:

- If the emoji naming format in Echat is consistent with the naming in Slack, the emoji transfers. This worked with 100% of the emoji I tested that had matching names.
- If the same emoji exists in both places but is named different; ex: `:thumbs-up:` vs `:+1:` they don't transfer. 
- If the emoji is custom, as expected it doesn't transfer.

In Slack
<img width="777" alt="2023-04-13_17-08-09" src="https://user-images.githubusercontent.com/10925636/231909164-677ce82c-bb53-4bd4-85a7-61c68b905365.png">

In Echat
<img width="1051" alt="2023-04-13_17-11-36" src="https://user-images.githubusercontent.com/10925636/231909469-b0ab1235-7ff8-479d-9357-0757bf883794.png">
